pallet/account-migration/src/lib.rs

```diff
@@ -457,14 +457,41 @@ pub mod pallet {
 				}
 
 				let staking_pot = darwinia_staking::account_id();
+				let now = <frame_system::Pallet<T>>::block_number();
+
 				<pallet_balances::Pallet<T> as Currency<_>>::transfer(
 					&to,
 					&staking_pot,
-					l.staked_ring + l.unstaking_ring.iter().map(|(r, _)| r).sum::<Balance>(),
+					l.staked_ring
+						+ l.unstaking_ring
+							.iter()
+							.filter_map(
+								|(amount, expired_time)| {
+									if expired_time <= &now {
+										None
+									} else {
+										Some(amount)
+									}
+								},
+							)
+							.sum::<Balance>(),
 					KeepAlive,
 				)?;
 
-				let sum = l.staked_kton + l.unstaking_kton.iter().map(|(k, _)| k).sum::<Balance>();
+				let sum = l.staked_kton
+					+ l.unstaking_kton
+						.iter()
+						.filter_map(
+							|(amount, expired_time)| {
+								if expired_time <= &now {
+									None
```

Comment:
Does the ledger here get updated?

Reply:
I just wanted to update this. Wait a min.

Reply:
Another thing I would like to check is that, for the migrations before this upgrate, the unstakings that moved to staking pot and stayed in ledger, can they claim their unstaking?

Do we need to fix for them?

Reply:
Yes. They can claim them immediately. @xiaoch05 did this yesterday.